document/JS.bs

```diff
@@ -0,0 +1,921 @@
+          1. Let |value| be [=ToWebAssemblyValue=](|v|, |globaltype|.valtype)
+          1. Assert: |globaltype|.mut is const, as verified by WebAssembly validation.
+          1. Let |store| be the current agent cluster's [=associated store=].
+          1. Let (|store|, |globaladdr|) be [=alloc_global=](|store|, |globaltype|, |value|).
+          1. Set the current agent cluster's [=associated store=] to |store|.
+          1. Let |externglobal| be 𝗀𝗅𝗈𝖻𝖺𝗅 |globaladdr|.
+          1. [=Append=] |externglobal| to |imports|.
+      1. If |externtype| is of the form 𝗆𝖾𝗆  |memtype|,
+          1. If |v| is not a {{Memory}} object, throw a {{LinkError}} exception.
+
+          Note: [=instantiate_module=] below will check the imported {{Memory}}'s size against the importing module's requirements.
+
+          2. [=Append=] |v| to |memories|.
+          1. Let |externmem| be the [=external value=] 𝗆𝖾𝗆  |v|.\[[Memory]].
+          1. [=Append=] |externmem| to |imports|.
+      1. Otherwise, |externtype| is of the form 𝗍𝖺𝖻𝗅𝖾 |tabletype|,
+          1. If |v| is not a {{Table}} instance, throw a {{LinkError}} exception.
+
+          Note: The table's length, etc is checked by [=instantiate_module=] below.
+
+          2. [=Append=] |v| to |tables|.
+          1. Let |tableaddr| be |v|.\[[Table]]
+          1. Let |externtable| be the [=external value=] 𝗍𝖺𝖻𝗅𝖾 |tableaddr|.
+          1. [=Append=] |externtable| to |imports|.
+          1. For each element |func| of |v|.\[[Values]],
+              1. If |func| is not null, append |func| to |funcs|.
+    1. Let (|store|, |instance|) be [=instantiate_module=](|store|, |module|, |imports|), up through step 16 of [=the instantiation algorithm=], but omitting execution of the start function.
+    1. If |instance| is [=error=], throw a {{LinkError}} exception, or other <a href="#errors">appropriate exception type</a>.
+    1. For each |tableaddr| in |instance|.𝗍𝖺𝖻𝗅𝖾𝖺𝖽𝖽𝗋𝗌,
+        1. If there is no element in |tables| whose |table|.\[[Table]] is |tableaddr|:
+            1. Let |table| be [=create a table object|a new table object=] from |tableaddr|.
+            1. [=Append=] |table| to |tables|.
+        1. Otherwise:
+            1. Let |table| be the element in |tables| where |table|.\[[Table]] is |tableaddr|.
+        1. For each index |i| from 0 to the length of |table|.\[[Values]] &minus; 1, do
+            1. Let |funcaddr| be [=read_table=](|store|, |tableaddr|, |i|).
+            1. Assert: |funcaddr| is not [=error=] because this read is in bounds.
+            1. If |funcaddr| is not empty,
+                1. If |funcs| contains any |func| where |func|.\[[ExportedAddress]] is |funcaddr|,
+                    1. Let |func| be that function object.
+                1. Otherwise:
+                    1. Let |func| be a [=create an exported function|a new exported function=] created from |funcaddr|.
```

Comment:
One place uses the text "a new exported function" and another "create an Exported Function"; best to be consistent.